### DISCUSSION TITLE
Unable to successfully perform Permanova analysis using the latest version 2.6-8

### DISCUSSION THREAD

**Discussion by jieyin2023** [vegandevs/vegan#694] [Q&A] [answered]
Dear author,
       I don't know why the output of the same code after updating Vegan is different. The new version of Vegan cannot output the impact of each treatments on community structure. only 
here is my code: permanova_results <- adonis2(formula = bray.dist ~ Experiment+ Soil +Genotype+Sample_type, data = merged_pc)
     The results of the old version is shown in the left figure, and the output of the new version is shown in the right figure. It is very important to me, please tell me why.
![Uploading old version.png…]()
![Uploading new verson.png…]()

**Comment by jarioksa** [comment, marked as answer]
The following pieces of information are of interest:

In NEWS for 2.6-8
* `adonis2()` now defaults to running an omnibus test of the model (`by = NULL`) instead of a sequential test of model terms (`by =
  "terms"`). This makes `adonis2()` more consistent with the default for related ordination methods.  See [#677](https://github.com/vegandevs/vegan/issues/677).

In `adonis2` help:
```
      by: ‘by = NULL’ will assess the overall significance of all terms
          together, ‘by = "terms"’ will assess significance for each
          term (sequentially from first to last), setting ‘by =
          "margin"’ will assess the marginal effects of the terms (each
          marginal term analysed in a model with all other variables),
          ‘by = "onedf"’ will analyse one-degree-of-freedom contrasts
          sequentially. The argument is passed on to ‘anova.cca’.
```
And these hint to issue #677.

That is, the default output was changed for a reason. You get the old output with argument `by="terms"` and marginal effects with `by="margin"`.
